enigma-swing/build.gradle

```
@@ -1,23 +1,36 @@
 plugins {
 	id 'application'
-	id 'com.github.johnrengelman.shadow' version '7.0.0'
+	id 'com.github.johnrengelman.shadow'
 }
 
+def flatLafNatives = [
+		"windows-arm64@dll",
+		"windows-x86@dll",
+		"windows-x86_64@dll",
+		"linux-x86_64@so",
+]
+
 dependencies {
 	implementation project(':enigma')
 	implementation project(':enigma-server')
 
 	implementation 'net.sf.jopt-simple:jopt-simple:6.0-alpha-3'
-	implementation 'com.formdev:flatlaf:2.6'
-	implementation 'com.formdev:flatlaf-extras:2.6' // for SVG icons
-	implementation 'de.sciss:syntaxpane:1.2.1'
+	implementation 'com.formdev:flatlaf:3.2.1'
+	implementation 'com.formdev:flatlaf-extras:3.2.1' // for SVG icons
+	implementation 'de.sciss:syntaxpane:1.3.0'
```

```suggestion
	implementation 'de.sciss:syntaxpane:1.2.1'
```
syntaxpane 1.3 breaks scrolling line numbers, as I discovered when trying to update for quilt enigma. you'll have to either:
- stay on 1.2 and wait for a fix
- move to [syntaxpain](https://github.com/quiltmc/syntaxpain), a minified version of the library I made so we could more easily maintain enigma-swing.

---

Yeah, ill just stick with the current version. Is there a bug report open for this issue?

---

Yup, it's here: <https://codeberg.org/sciss/SyntaxPane/issues/38>